src/properties/components/Properties.tsx

```diff
@@ -25,21 +25,29 @@ import { tryFindByUUID } from '../../core/util/clone';
 import { ExamplePropertiesService } from '../propertiesService';
 
 const propertiesService = new ExamplePropertiesService();
+
 const getProperties = (
   uiElement: EditorUISchemaElement | undefined,
   schema: SchemaElement | undefined
 ) => {
-  if (!uiElement || !schema) {
+  if (!uiElement) {
     return undefined;
   }
   const linkedSchemaUUID = uiElement.linkedSchemaElement;
-  if (!linkedSchemaUUID) {
-    return undefined;
-  }
-  const elementSchema = tryFindByUUID(schema, linkedSchemaUUID);
+  const elementSchema =
+    linkedSchemaUUID && schema
+      ? tryFindByUUID(schema, linkedSchemaUUID)
+      : undefined;
   return propertiesService.getProperties(uiElement, elementSchema);
 };
 
+const canUpdateUiSchemaElement = (
+  uiElement: EditorUISchemaElement | undefined,
+  updatedProperties: any
+): boolean => {
+  return !isEqual(uiElement, updatedProperties);
+};
+
```

Comment:
Here we then need to compare against the extracted element. This will also make the check much faster as this compares the whole ui element tree at the moment.